test/Concurrency/async_sequence_syntax.swift

```diff
@@ -33,9 +33,9 @@ func missingTryInBlock<T : AsyncSequence>(_ seq: T) {
 
 @available(macOS 9999, iOS 9999, watchOS 9999, tvOS 9999, *)
 func missingAsyncInBlock<T : AsyncSequence>(_ seq: T) { 
-  execute { // expected-error{{invalid conversion from 'async' function of type '() async -> Void' to synchronous function type '() -> Void'}}
+  execute { // expected-error{{cannot pass function of type '() async -> Void' to parameter expecting synchronous function type}}
     do { 
-      for try await _ in seq { }
+      for try await _ in seq { } // expected-note {{'async' in a closure that does not support concurrency}}
```

Comment:
I'm not sure about the wording here. `async` doesn't explicitly show up in this expression, so that might not be totally clear. I don't have a case for it yet, but a case for the `for await` construction might belong in the `async_access_without_await` note in `DiagnosticsSema.def`. That note should probably be renamed to something like `async_access_note` or something since it's just noting things that are async.

I'm fine with this as-is, but just my initial thoughts.

Reply:
I'm happy to change a note, I'm just not sure how to phrase it. Would something like `asynchronous operation in closure that does not support concurrency` be better?

Reply:
Yeah, I think that makes it clearer that it's not the `async` keyword that they should be looking for. Does the note point at the `for` or the `await`?

Reply:
It points to a `for` since that's the node "await" is attached to, I can detect that and make it point to `await` itself if that's better.

Reply:
I think it's confusing for the error to say that the closure is asynchronous and then to emit a note saying that the closure does not support concurrency. I think the note should say something more along the lines of "async inferred from here" (but with better wording)

Reply:
Settled on `'async' inferred from asynchronous operation used here` for this.

Reply:
Updated!